WalletWasabi/Stores/IndexStore.cs

```diff
@@ -195,10 +203,13 @@ private void MigrateToSqliteNoLock(CancellationToken cancel)
 			Logger.LogError(ex);
 
 			SqliteConnection.ClearAllPools();
+			IndexStorage.Dispose();
```

Comment:
I don't know what SqliteConnection.ClearAllPools() would do _after_ and IndexStorage.Dispose(), but I do know that the ClearAllPools() won't crash itself.
I really hope that we don't need to mess with GC by now (2024) and the SQLite's close and dispose does the job (since that's the exact cause for its existence).

Reply:
If we see more crashes like this even after the PR, we might go an extra round with it, but I expect that we won't have this issue.
To be honest, we could even just add a heads up in the documentation that in such case delete the Main folder manually while the Wasabi is not running and restart. It's a one time annoyance for the user instead of stressing on what is going on.

Reply:
Flipped the call order.

Reply:
> IndexStorage.Dispose() would dispose stuff correctly as is customary and SqliteConnection.ClearAllPools() would remove any lingering connections/whatever. Doing in the reverse order can be done but I would expect an explanation as it seems less obvious why it was done that way.

At first glance, I would have done in the same order as the first commit. (1st Clear Connections, 2nd Dispose the whole object)

@kiminuo Could you please tell me why the reserved order is better? For me, it feels weird that we first dispose the object, and only after that we close the disposed object's connection. 
Won't we run into `ObjectDisposedException`? or the connection pool is handled by another component?

I checked the stackoverflow link you sent and it seems like people first close the connection and then force the GC to do its work (dispose).

Reply:
Thank you for the explanation @kiminuo!